### ISSUE TITLE
Can the SDDManage create SDD from CNF file and at the same time follow a predefined vtree?

### ISSUE THREAD

**Issue by pangzhan27** [wannesm/PySDD#23]
Following the example you offered, i have done several trials. Below is one example.

####################
vtree = Vtree.from_file(bytes("test_sdd/test/2.vtree".encode()))
sdd = SddManager.from_vtree(vtree)
file_name = Path('./test_sdd/test/2.cnf')
root = sdd.read_cnf_file(bytes(file_name))
############################

The problem is that even the SDD file is created, its corresponding vtree structure is not the same as i provide. 
So, any ideas how to fix the vtree structure and then generate the SDD file.

**Comment by VincentDerk** [comment]
I took a look at the code and found the following:

1. read_cnf_file(...) calls fnf_to_sdd(...)
```python
    def read_cnf_file(self, filename):
        """Replace the SDD by an SDD representing the theory in the given CNF file."""
        cdef Fnf cnf = Fnf.from_cnf_file(filename)
        return self.fnf_to_sdd(cnf)
```

2. `fnf_to_sdd` uses `vtree_search_mode` to determine whether to automatically optimise the sdd (`fnf_to_sdd_auto`) or not (`fnf_to_sdd_manual`). Note that optimising the SDD means changing the vtree. I guess you do not want this.
```cython
SddNode* fnf_to_sdd(Fnf* fnf, SddManager* manager) {
  SddNode* test = degenerate_fnf_test(fnf,manager);
  if (test != NULL) return test;
  SddCompilerOptions* options = sdd_manager_options(manager);

  if(options->vtree_search_mode < 0) {   <------------------- here
    sdd_manager_auto_gc_and_minimize_on(manager);
    return fnf_to_sdd_auto(fnf,manager);
  } else {
    sdd_manager_auto_gc_and_minimize_off(manager);
    return fnf_to_sdd_manual(fnf,manager);
  }
}
```
3. `vtree_search_mode` is stored in `CompilerOptions`.

TL;DR:
Before creating the SDD from file via `read_cnf_file(file)`, first set the options of the Sdd Manager.
```python
from pysdd.sdd import Vtree, SddManager, WmcManager, FNf, CompilerOptions
vtree = Vtree.from_file(bytes("test_sdd/test/2.vtree".encode()))
sdd = SddManager.from_vtree(vtree)
# new
new_options = CompilerOptions(vtree_search_mode=0)
sdd.set_options(new_options)
# 
file_name = Path('./test_sdd/test/2.cnf')
root = sdd.read_cnf_file(bytes(file_name))
```

**Comment by pangzhan27** [comment]
I see. I have tried in your way. It works.
Many thanks!
